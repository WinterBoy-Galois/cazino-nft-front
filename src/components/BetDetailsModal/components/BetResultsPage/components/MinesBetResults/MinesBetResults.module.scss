@import '../../../../../../styles/shared';

//.container {
//  display: flex;
//  flex-wrap: wrap;
//  justify-content: center;
//}

.container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
  grid-auto-rows: 1fr;
  grid-column-gap: 0;
  grid-row-gap: 0;
  transition: all 0.3s ease-out;
  justify-items: center;
}
.field {
  position: relative;
  width: 54px;
  height: 54px;
  opacity: 0.5;
  margin: 0.3rem;
  border-radius: 0.5rem;

  @include media-breakpoint-up(xl) {
    width: 70px;
    height: 70px;
  }

  &__index {
    position: absolute;
    width: 24px;
    height: 24px;
    top: 0.3rem;
    left: 0.3rem;
    border-radius: 50%;
    background-color: rgba($white, 0.8);
    color: #000;
    display: flex;
    justify-content: center;
    align-items: center;
    font-weight: 600;
    font-size: 14px;

    @include media-breakpoint-up(xl) {
      width: 28px;
      height: 28px;
      font-size: 16px;
    }
  }

  &--opened {
    opacity: 1;
  }

  &--loss {
    background-color: rgba($negative, 0.5);
    padding: 0.2rem;

    @include media-breakpoint-up(xl) {
      padding: 0.3rem;
    }
  }

  &--win {
    background-color: rgba($positive, 0.5);
    padding: 0.4rem;

    @include media-breakpoint-up(xl) {
      padding: 0.6rem;
    }
  }
}
