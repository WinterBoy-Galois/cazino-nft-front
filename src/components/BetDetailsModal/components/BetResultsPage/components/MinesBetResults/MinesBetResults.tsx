import React from 'react';
import styles from './MinesBetResults.module.scss';
import Mine from '../../../../../../assets/images/games/mines/mines-boom.svg';
import Cash from '../../../../../../assets/images/games/mines/mines-bitcoin.svg';

interface IProps {
  fieldCount?: number;
  minePositions: number[];
  openedFields: number[];
}

const MinesBetResults: React.FC<IProps> = ({ fieldCount = 25, minePositions, openedFields }) => {
  return (
    <div className={styles.container}>
      {Array.from(new Array(fieldCount)).map((_, fieldIndex) => (
        <div
          key={fieldIndex}
          className={`${styles.field} ${
            openedFields.includes(fieldIndex) ? styles['field--opened'] : ''
          } ${minePositions.includes(fieldIndex) ? styles['field--loss'] : styles['field--win']}`}
        >
          <span className={styles.field__index}>{fieldIndex}</span>
          {minePositions.includes(fieldIndex) ? (
            <img src={Mine} alt="Mine" />
          ) : (
            <img src={Cash} alt="Cash" />
          )}
        </div>
      ))}
    </div>
  );
};

export default MinesBetResults;
