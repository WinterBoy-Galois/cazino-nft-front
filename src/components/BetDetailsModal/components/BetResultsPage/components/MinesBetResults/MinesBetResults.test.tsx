import React from 'react';
import { render } from '@testing-library/react';
import MinesBetResults from '.';

describe('MinesBetResults', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <MinesBetResults
        fieldCount={24}
        minePositions={[4, 10, 22]}
        openedFields={[0, 5, 10, 15, 24]}
      />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
