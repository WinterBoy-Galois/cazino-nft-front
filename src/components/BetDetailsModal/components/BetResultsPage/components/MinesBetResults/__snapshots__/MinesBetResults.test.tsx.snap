// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinesBetResults should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="field field--opened field--win"
        >
          <span
            class="field__index"
          >
            0
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            1
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            2
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            3
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--loss"
        >
          <span
            class="field__index"
          >
            4
          </span>
          <img
            alt="Mine"
            src="mines-boom.svg"
          />
        </div>
        <div
          class="field field--opened field--win"
        >
          <span
            class="field__index"
          >
            5
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            6
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            7
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            8
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            9
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field field--opened field--loss"
        >
          <span
            class="field__index"
          >
            10
          </span>
          <img
            alt="Mine"
            src="mines-boom.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            11
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            12
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            13
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            14
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field field--opened field--win"
        >
          <span
            class="field__index"
          >
            15
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            16
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            17
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            18
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            19
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            20
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            21
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
        <div
          class="field  field--loss"
        >
          <span
            class="field__index"
          >
            22
          </span>
          <img
            alt="Mine"
            src="mines-boom.svg"
          />
        </div>
        <div
          class="field  field--win"
        >
          <span
            class="field__index"
          >
            23
          </span>
          <img
            alt="Cash"
            src="mines-bitcoin.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="field field--opened field--win"
      >
        <span
          class="field__index"
        >
          0
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          1
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          2
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          3
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--loss"
      >
        <span
          class="field__index"
        >
          4
        </span>
        <img
          alt="Mine"
          src="mines-boom.svg"
        />
      </div>
      <div
        class="field field--opened field--win"
      >
        <span
          class="field__index"
        >
          5
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          6
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          7
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          8
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          9
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field field--opened field--loss"
      >
        <span
          class="field__index"
        >
          10
        </span>
        <img
          alt="Mine"
          src="mines-boom.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          11
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          12
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          13
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          14
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field field--opened field--win"
      >
        <span
          class="field__index"
        >
          15
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          16
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          17
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          18
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          19
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          20
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          21
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
      <div
        class="field  field--loss"
      >
        <span
          class="field__index"
        >
          22
        </span>
        <img
          alt="Mine"
          src="mines-boom.svg"
        />
      </div>
      <div
        class="field  field--win"
      >
        <span
          class="field__index"
        >
          23
        </span>
        <img
          alt="Cash"
          src="mines-bitcoin.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
